### ISSUE TITLE
Incorrect "inline text color highlighting"

### ISSUE THREAD

**Issue by Monnoroch** [Monnoroch/ColorHighlighter#391]
Mane people report that when using "inline highlighting > text" the background is incorrect:

https://user-images.githubusercontent.com/3880497/31282228-78d7df52-aae4-11e7-8d8d-804d4520307a.png

https://user-images.githubusercontent.com/722614/31081941-c180e2b2-a78d-11e7-9bde-5c766720e0ab.png

https://image.prntscr.com/image/Atslm0AwT9WhU0Itty-cEw.jpeg

I can't reproduce this myself, so I need help on this.

#389, #378, #382.

**Comment by lichtamberg** [comment]
OSX Sierra, ColorHighlighter 8.0.9

Sublime:
![bildschirmfoto 2017-10-09 um 16 15 55](https://user-images.githubusercontent.com/10075/31342566-7895d122-ad0d-11e7-9494-30ecd69c0c3d.png)

User-Settings:
```json
{
    "enabled": true,
    "style": "default",
    "icons": true,
    "ha_style": "solid",
    "icons_all": false
}

```

![bildschirmfoto 2017-10-09 um 16 19 58](https://user-images.githubusercontent.com/10075/31342652-bc9fd872-ad0d-11e7-84be-252f8ba36e99.png)

**Comment by Monnoroch** [comment]
@lichtamberg your user settings are from the old version and do not work on the new one.

**Comment by lichtamberg** [comment]
I see, I just disabled the block display and now it doesnt work anymore at all -.-

**Comment by lichtamberg** [comment]
I removed now my whole user config for color highligher and have the same problem again now:

![bildschirmfoto 2017-10-09 um 17 27 07](https://user-images.githubusercontent.com/10075/31345781-1b2d918c-ad17-11e7-921d-8ce3fb8dd2a8.png)

**Comment by Monnoroch** [comment]
@lichtamberg what is your color scheme called?

**Comment by lichtamberg** [comment]
```json
"color_scheme": "Packages/RailsCasts Colour Scheme/RailsCastsColorScheme.tmTheme",
```

But you are right, it looks like it's related to the color scheme.. When I chose a different one, it looks like its working better.. Will test more..

**Comment by Monnoroch** [comment]
@lichtamberg wait, you posted the color scheme settings. Some plugins override the scheme for specific views. Can you screenshot the color scheme picker view while having the file with colors open.

**Comment by lichtamberg** [comment]
![bildschirmfoto 2017-10-09 um 17 38 46](https://user-images.githubusercontent.com/10075/31346255-c6eba512-ad18-11e7-938a-42f6fa2d96b5.png)
![bildschirmfoto 2017-10-09 um 17 39 36](https://user-images.githubusercontent.com/10075/31346293-e179a050-ad18-11e7-99f0-01e69f3a0632.png)

**Comment by lichtamberg** [comment]
LOL. It work's now, but the only thing which I changed was, that I just copied the whole ColorHighlighter-Config into the user config.

**Comment by Monnoroch** [comment]
@lichtamberg you have a sublime linter enabled: https://github.com/Monnoroch/ColorHighlighter#inline-color-highlighting-works-incorrectly.

**Comment by Monnoroch** [comment]
Can you please test it with SL disabled?

**Comment by lichtamberg** [comment]
Problem is now, that it's working :D
If I have the same problem again, I will disable SL and report it if I continue to have problems. Thank you very much for taking time and your great plugin!

**Comment by Monnoroch** [comment]
@lichtamberg my hypothesis is that SL and CH compete for the color scheme and break each other. If you have the problem again, please try to disable SL and check it that way.

**Comment by isaumya** [comment]
I'm having very weird color highlighting in Windows as well. Screenshot attached. I have no idea if this has any way fix. Take a look.
![2017-10-15_22-15-23](https://user-images.githubusercontent.com/2124755/31586965-a7e19a4a-b1f6-11e7-86bd-0022ac976dee.jpg)
I have no idea where that orange background is coming from.
BTW I'm using the latest Sublime Text on Windows 10

**Comment by Monnoroch** [comment]
@isaumya what plugins do you use besides CH?

**Comment by isaumya** [comment]
@Monnoroch in terms of theme I use the Material Theme.  Along with A File Icon, Bracket Highlighter.
The rest packages I use has nothing to do with interface. It is mostly for coding.

**Comment by Monnoroch** [comment]
@isaumya this is the most likely reason: https://github.com/Monnoroch/ColorHighlighter#inline-color-highlighting-works-incorrectly. Could you please test CH without any plugins enabled (and a ST reload) and see if the problem persists? If it doesn't, please find the incompoatible plugin so I can add it to the readme ar an issue. This would be of great help!

**Comment by isaumya** [comment]
@Monnoroch Hi, I was able to make it work just changing a small settings in the color highlighter:

![2017-10-15_23-44-56](https://user-images.githubusercontent.com/2124755/31587652-063b2eec-b203-11e7-9e40-c78ddfdec098.jpg)

Earlier it was `Text` and I have changed it to `Filled`.

**Comment by Monnoroch** [comment]
@isaumya good, but that doesnt help me with debugging the problem. Can you change it back now? Will it work?

**Comment by isaumya** [comment]
No. if I change it back to `Text` it gets messed up. I think it is an incompatibility with material theme. I'm happy with changing the `Text` to `Filled`. :)

**Comment by Monnoroch** [comment]
@isaumya can you please test it with the default Monokai? Does it work correctly then?

**Comment by isaumya** [comment]
@Monnoroch yes I've tried with the default theme and it work fine with that. But as I can't consider using anything but the Material theme (because I'm deeply in love with it), so for the me `Filled` thing is just work fine.

**Comment by Monnoroch** [comment]
@isaumya ok, thanks, good to know that not all themes are compatible!

**Comment by jaegz** [comment]
I too was getting the orange highlight but the text was showing the correct colors.
@isaumya's fix worked for me too.

`Tools > Color Highlighter > Color highlighters > Highlight colors in all text > Inline highlighting style` and then changed `Text` to `Filled` and now it works perfectly. Changing it back does not work.

![colorhighlighter-orangehighlight](https://user-images.githubusercontent.com/7094801/31623202-b8c77e8a-b26c-11e7-9eae-b3d18a20c844.jpg)
![colorhighlighter-orangehighlight-fixed](https://user-images.githubusercontent.com/7094801/31623203-b8d6cfb6-b26c-11e7-9a02-a54e10a320d4.jpg)

**System**
Windows 7 Pro 64bit
Sublime Text 3 Version 3.0, Build 3143

**Sublime Settings**
Theme: Seti UI 
Color Scheme: Blackboard (SL)

**Packages**
ApplySyntax
Babel
Color Highlighter
EJS
Emmet
Package Control
PyV8
Sass
Seti_UI
Seti_UX
SublimeLinter
SublimeLinter-jshint
TernJS
Themr

**Comment by Monnoroch** [comment]
@jaegz can you also test it with the defaut color scheme and tell me if it works or not?

**Comment by Monnoroch** [comment]
@jaegz actually, it won't, you have a problem because of https://github.com/Monnoroch/ColorHighlighter#inline-color-highlighting-works-incorrectly.

**Comment by jaegz** [comment]
@Monnoroch thanks for the reference. So in my case, running the Blackboard (SL) color scheme is where the conflict arises as Monokai (default) did work for me along with the highlight `Text` setting

![colorhighlighter-monokai default](https://user-images.githubusercontent.com/7094801/31623795-9a131b5a-b26e-11e7-9f62-cf0345d1c60a.jpg)

**Comment by mikehdt** [comment]
Definitely some unusual behaviours with SublimeLinter. Disabling all the SL plugins doesn't seem to stop the behaviour of making copied (SL) schemes. I guess it has to be uninstalled?

What's weird though, is I've had them coexisting peacefully in the past - not sure whether it's newer Sublime builds (using 3150 on High Sierra) or the CH 8.0 changes or something else. I can't seem to make them be peaceful again. It'd be nice if Sublime had a better way for plugins to mix in their theme overrides rather than making loads of copies of colour schemes :S

 Here it seems to invert the colour underlines, although the gutter swatch is correct:

![screen shot 2017-10-19 at 10 56 10 am](https://user-images.githubusercontent.com/4432939/31748088-6d24a762-b4bc-11e7-97b5-7d79afd03bd8.png)

Clicking on colours doesn't appear happy at all though, it just seems to use the default colour scheme highlight.

**Comment by Monnoroch** [comment]
@mikehdt there is a way to mix them: it's called use something other than "inline" highlighting in CH :) However it's not as pretty to some people.

**Comment by mikehdt** [comment]
@Monnoroch I'm not sure if there was an update to CH or whether the most recent Sublime build (3153) has done something, but the colour highlighting is working well again - correct colours, correct highlighting on click :D

**Comment by Monnoroch** [comment]
@mikehdt I'm tolally fine with taking the credit!

**Comment by isaumya** [comment]
I can also confirm this. Now with `Filled` selected there is no issue whatsoever. Recently both Sublime Text and the Material Theme got updated. Not sure whose update fixed this issue. But whoever it is, thank you for fixing it. 🙏
